packages/oidc/src/refresh/refreshGrant.ts

```diff
@@ -62,6 +85,11 @@ export async function refresh(
         `${client.clientId}:${client.clientSecret}`
       )}`,
     };
+  } else if (client.clientId !== undefined && isValidUrl(client.clientId)) {
```

Comment:
Should line 80 also do:
```
if (client.clientId !== undefined && client.clientSecret !== undefined) {
```

Reply:
I approached this somewhat differently with f57f1282: the absence of a client_id is not something we support, so we might as well throw an error. The issue should actually be caught when looking up the credentials from storage, which is why there was initially no check to that intent.